### ISSUE TITLE
[BUG] iPhone 6 don't support bigInt

### ISSUE THREAD

**Issue by alanlawlor** [clawfire/covid19-passbook-generator#54]
**Describe the bug**
Using iPhone 6+ on ios12.5
Tried on Safari and Chrome browsers

**To Reproduce**
1. Go to https://covid19passbook.netlify.app/
2. Tap on “Start scanning” button
3. Nothing happens
4. Tap on “import picture” button
5. Nothing happens

In the case of chrome, I checked iOS settings and Chrome has access to Camera and to Photos (read and write)

**Expected behavior**
After tapping on button, expect to see a camera window open or a gallery of my photos to select.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: iPhone6 plus
 - OS: iOS 12.5.3
 - Browser: both chrome & safari have same problem 

**Additional context**

**Comment by clawfire** [comment]
Hi there 😃 

That sounds strange; accessing the camera require using the site in a plain browser and not an "in-app" browser like the one used in Messenger, Facebook, Twitter, ... That's a limitation both from the app creator (for Facebook, Messenger) and Apple (for Twitter, Instagram, ...). But it seems to be your case, right?

Btw, pushing a passbook in the Apple wallet requires using Safari so far. Didn't find how to make it work for another browser (since we are not generating something on a server, all your data staying in your browser, but it means you're not "downloading" something per se, it's all happening in memory)

We are working on some improvements to detect and tell users what to do when they are in that case #25

**Comment by clawfire** [comment]
Hey there. I pushed a lot of changes recently, could you try it again ? 😃

**Comment by alanlawlor** [comment]
Hi Thibault

Still not working on Safari for iOS 12.5

But as per my Twitter reply, I managed to work around it


Alan

On 31 Jul 2021, at 02:31, Thibault Milan ***@***.******@***.***>> wrote:


Hey there. I pushed a lot of changes recently, could you try it again ? 😃

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fclawfire%2Fcovid19-passbook-generator%2Fissues%2F54%23issuecomment-890272459&data=04%7C01%7Calawlor%40dunnes-stores.ie%7C13b53346329d4435269808d953c2e350%7C5858e8286aa549d48979fb3edbe6aeac%7C0%7C0%7C637632918785865712%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=RN5DHx7LzO%2Bya9mNCu%2FVzr%2FCDJSIBKpYLtsk%2BvZyjRA%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FACUPDNWQOJ2SQIUUZMB5DY3T2NG6DANCNFSM5AK36FFA&data=04%7C01%7Calawlor%40dunnes-stores.ie%7C13b53346329d4435269808d953c2e350%7C5858e8286aa549d48979fb3edbe6aeac%7C0%7C0%7C637632918785865712%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=c6J0vsZQbZnvkoUJNb18C8ZaKxPjWLbNi0l1rUN2648%3D&reserved=0>.

**************************************************** **************************************************** DISCLAIMER Any opinions expressed in this email are those of the individual and not necessarily the Company. This email and any files transmitted with it, including replies and forwarded copies (which may contain alterations) subsequently transmitted from the Company, are confidential and may be privileged and are solely for the use of the intended recipient. If you are not the intended recipient or the person responsible for delivering to the intended recipient, be advised that you have received this email in error and that any use is strictly prohibited. If you have received this email in error please notify the Business Systems Department Helpdesk by telephone on +353 (0)1 6112714 or via email to ***@***.***, including a copy of this message. Please then delete this email and destroy any copies of it. **************************************************** ****************************************************

**Comment by clawfire** [comment]
Seems we have an issue with iPhone 6.

**Comment by clawfire** [comment]
Found why. 
I'm using a javascript feature that is supported by Safari since ios 14.4 but iPhone 6 are blocked to ios 12 max :( 

This seems to be require to be able to use the cryptographic functions to decode the content of the qrcode (not the qr code itself but the content isn't readable like this)

https://caniuse.com/bigint

**Comment by clawfire** [comment]
TODO : 
- [ ] Add the feature detection of bigInt (or ios 14 as requirement) with a screen saying it's impossible to use

**Comment by clawfire** [comment]
Found out the bug was caused by the lack of support of BigInt in Safari before version 14. And there's no polyfill we can put in place to support this. BigInt represents big integer numbers in memory, required for cryptographic manipulation such as decoding the QR code.

I implemented a detection feature of lack of this functionality and warned the user with a message + prevent to discard the message (since the app will be totally unusable anyway).

iPhones 6 and 6+ are all stuck in iOS12 max, which isn't supported.
IPhones after 6 (including 6s) can upgrade to iOS14 and should ti get the app working.
